app/client/components/delivery/records/deliveryAddressStep.tsx

```diff
@@ -0,0 +1,582 @@
+    setFormErrors({
+      addressLine1: isFormValidResponse.addressLine1,
+      town: isFormValidResponse.town,
+      postcode: isFormValidResponse.postcode,
+      country: isFormValidResponse.country
+    } as FormValidationResponse);
+
+    if (isFormValidResponse.isValid && acknowledgementChecked) {
+      props.setAddressValidationState(true);
+      setStatus(Status.CONFIRMATION);
+    } else {
+      setStatus(Status.VALIDATION_ERROR);
+    }
+  };
+
+  const renderDeliveryAddressForm = (
+    allProductDetails: MembersDataApiItem[]
+  ) => {
+    const contactIdToArrayOfProductDetail = getValidDeliveryAddressChangeEffectiveDates(
+      allProductDetails
+        .filter(isProduct)
+        .filter(product => product.subscription.readerType !== "Gift")
+    );
+
+    const addressChangeAffectedInfoArray = addressChangeAffectedInfo(
+      contactIdToArrayOfProductDetail
+    );
+
+    setAddressChangeInformation(
+      [
+        ...addressChangeAffectedInfoArray.map(
+          element =>
+            `${element.friendlyProductName} subscription (${
+              element.subscriptionId
+            })${
+              element.effectiveDate
+                ? ` as of front cover dated ${element.effectiveDate.format(
+                    "dddd Do MMMM YYYY"
+                  )}`
+                : ""
+            }`
+        ),
+        "",
+        `(as displayed on confirmation page at ${moment().format(
+          "H:mm:ss zz [on] Do MMMM YYYY"
+        )})`
+      ].join("\n")
+    );
+
+    const productsAffected: ProductDescriptionListKeyValue[] = convertToDescriptionListData(
+      addressChangeAffectedInfoArray
+    );
+
+    const subscriptionNames = Object.values(contactIdToArrayOfProductDetail)
+      .flatMap(flattenEquivalent)
+      .map(productDetail => {
+        const friendlyProductName = ProductTypes.contentSubscriptions.mapGroupedToSpecific?.(
+          productDetail
+        ).friendlyName;
+        return `${friendlyProductName}`;
+      });
+
+    return (
+      <>
+        {productsAffected.length > 1 && (
+          <p
+            css={css`
+              ${textSans.medium()};
+              background-color: ${palette.neutral[97]};
+              padding: ${space[5]}px ${space[5]}px ${space[5]}px 49px;
+              margin-bottom: 12px;
+              position: relative;
+            `}
+          >
+            <i
+              css={css`
+                width: 17px;
+                height: 17px;
+                position: absolute;
+                top: ${space[5]}px;
+                left: ${space[5]}px;
+              `}
+            >
+              <InfoIconDark fillColor={palette.brand[500]} />
+            </i>
+            Please note that changing your address here will update the delivery
+            address for all of your subscriptions.
+          </p>
+        )}
+        <form
+          action="#"
+          onSubmit={handleFormSubmit(subscriptionNames, productsAffected)}
+        >
```

Comment:
is there a reason this whole form component can't be re-used from the delivery address update flow?

Reply:
We will write a follow up pr to look into this 👍